### PR TITLE
3 add s2 library

### PR DESCRIPTION
**Pull request by ankitgit**

### CODE REVIEW

File: vcpkg-registry/ports/s2/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
+set(VERSION v2.6.0)
```
Comment (digiou) [inline vcpkg-registry/ports/s2/portfile.cmake:1]: Are we using the version string at all? Looks like it's mismatched too (2.6 vs 0.1)

**Review by digiou — APPROVED** [review]
LGTM, minor comment below.

File: vcpkg-registry/ports/s2/portfile.cmake (hunk 1 above)
Comment (ankitgit) [reply vcpkg-registry/ports/s2/portfile.cmake:1]: thanks for pointing it out

**Review by varpande — APPROVED** [review]
LGTM. I left a comment about abseil, but not entirely sure if that has an affect on the build process or not.

File: vcpkg-registry/ports/s2/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "s2",
+  "version": "0.10.0",
+  "description": "S2 is a library for spherical geometry that aims to have the same robustness, flexibility, and performance as the very best planar geometry libraries.",
+  "homepage": "https://s2geometry.io",
+  "license": "Apache-2.0",
+  "supports": "!windows",
+  "dependencies": [
+    "abseil",
```
[the remark this answers is not in the document]
Comment (ankitgit) [reply vcpkg-registry/ports/s2/vcpkg.json:9]: I manually tested this yesterday and found that the binaries work and the code compile without issue. Initially, one const from abseil was having ambiguous definitions. In one version it is an enum and in another, it is a bool. So I created a patch to rename it.

**Pull request merged**